consensus/hotstuff/persister/persister.go

```diff
@@ -36,34 +35,38 @@ func New(db *badger.DB, chainID flow.ChainID) (*Persister, error) {
 }
 
 // NewReader returns a new Persister as a PersisterReader type (only read methods accessible).
-func NewReader(db *badger.DB, chainID flow.ChainID) (hotstuff.PersisterReader, error) {
+func NewReader(db storage.DB, chainID flow.ChainID) (hotstuff.PersisterReader, error) {
 	return New(db, chainID)
 }
 
 // GetSafetyData will retrieve last persisted safety data.
 // During normal operations, no errors are expected.
 func (p *Persister) GetSafetyData() (*hotstuff.SafetyData, error) {
 	var safetyData hotstuff.SafetyData
-	err := p.db.View(operation.RetrieveSafetyData(p.chainID, &safetyData))
+	err := operation.RetrieveSafetyData(p.db.Reader(), p.chainID, &safetyData)
 	return &safetyData, err
 }
 
 // GetLivenessData will retrieve last persisted liveness data.
 // During normal operations, no errors are expected.
 func (p *Persister) GetLivenessData() (*hotstuff.LivenessData, error) {
 	var livenessData hotstuff.LivenessData
-	err := p.db.View(operation.RetrieveLivenessData(p.chainID, &livenessData))
+	err := operation.RetrieveLivenessData(p.db.Reader(), p.chainID, &livenessData)
 	return &livenessData, err
 }
 
 // PutSafetyData persists the last safety data.
 // During normal operations, no errors are expected.
 func (p *Persister) PutSafetyData(safetyData *hotstuff.SafetyData) error {
-	return operation.RetryOnConflict(p.db.Update, operation.UpdateSafetyData(p.chainID, safetyData))
+	return p.db.WithReaderBatchWriter(func(rw storage.ReaderBatchWriter) error {
```

Comment:
We are changing from updating with badger transaction to badger batch updates.

Since this only update one single data, there should be no difference.